README.md

```diff
@@ -68,12 +83,16 @@ Alternatively, you can manually execute the validations by running `pre-commit r
 
 JCCLI's test suite consists of unit tests and integration tests. The integration tests depend on the environment
 variable `JC_API_KEY`, which should be a Jumpcloud API key corresponding to a blank Jumpcloud instance which can be used
-for testing purposes.
+for testing purposes. The test suite also requires the `HOME` var to be set to the working directory, so that running
+JCCLI doesn't unintentionally overwrite the user's existing config file (unless tests are running on Travis CI).
 
 We use [Travis-CI](https://travis-ci.org/) to automate our testing. This repo's Travis configuration is set up to run
 the unit test suite (in `unit_tests/`) on every pull request and push, and to run the integration test suite (in
 `integration_tests/`) only on a push.
 
+Currently, the test suite expects the user running the tests to have a config file `.jccli.ini` in their home directory,
+with a section named `[jccli-dev-testing]`. This section currently does not need to have any content.
```

Comment:
I don't understand why you would have a section without any content?  If the user doesn't provide a key then how does the integration tests know which JC host to target?

Reply:
Do these tests run against the same jumpcloud endpoints as non-test calls, just segregated by a different api key (the host endpoint being embedded into jcapi-python)?

It looks like both the JC_API_KEY environment variable and a config file with an (empty) jccli-dev-testing are required to run integration tests. I can see the environment variable overriding a config key, but why not allow the key in the section then if the config section is required for testing?

Reply:
> Do these tests run against the same jumpcloud endpoints as non-test calls, just segregated by a different api key (the host endpoint being embedded into jcapi-python)?

Yes. Is there some way we can make this clearer in the README?

> but why not allow the key in the section then if the config section is required for testing?

You're absolutely right. I'll enable both.

The one possibility we really want to avoid is some developer who has a production API key in their `[DEFAULT]` section, and no key overriding that in the `[jccli-dev-testing]` section. If said developer were to accidentally run our CLI without setting the `JC_API_KEY` env var, they would end up running integration tests on their production JC account. In order to prevent this:

What if we have a ban storing API key in `[DEFAULT]` section of `.jccli.ini`? If we do, should I make that a special rule enforced before integration testing, or would it be a good policy for all usages?

Reply:
Having it use a specific testing profile makes sense to me, I just thought it was strange that the api key wasn't supported in there. If it was supported in the jccli-dev-testing profile, wouldn't that be enough to avoid someone accidentally running against the DEFAULT key (since the tests always us env variable or the specific test profile)?

Reply:
No

```
[DEFAULT]
key=123-production-key
format=json
indent=4

[jccli-dev-testing]
key=456-test-key
indent=2
```
If the user's `.jccli.ini` looks like the above, then `jccli --profile jccli-dev-testing <some-command-here>` will run with `API key = 456-test-key` (which is what we want 👍)

```
[DEFAULT]
key=123-production-key
format=json
indent=4

[jccli-dev-testing]
indent=2
```

However, if the user's `.jccli.ini` looks like the above, then `jccli --profile jccli-dev-testing <some-command-here>` will run with `API key = 123-production-key`.

My proposed solution is to have `jccli` look at whether there is a `key` defined in the `[DEFAULT]` section

Reply:
@zaro0508 would love to hear your thoughts as well

Reply:
> However, if the user's .jccli.ini looks like the above, then jccli --profile jccli-dev-testing <some-command-here> will run with API key = 123-production-key

Why is it falling back to DEFAULT key when a specific profile is specified?  The DEFAULT is only when you don't pass in a profile at all (i..e    jccli <some-command-here>).  It should not fallback to the default profile if a specific profile is passed in.

Also I don't think it makes sense to prevent tests from running if there is a `key` in `[DEFAULT]` section.

Reply:
That's how `ConfigParser` works. Any field that isn't present in the selected `[section]` gets defaults to whatever its value is in `[DEFAULT]`.

We probably want to take advantage of this behavior for stuff like formatting, e.g.:

```
[DEFAULT]
format=yaml
indent=2

[sage-admin]
key=789-sage-jc-key
format=json

[other-account]
key=987-some-key
indent=4
```

In the above example, running with `--profile sage-admin` will run with the API key `789-sage-jc-key`, and formatted in JSON, but the indentation level will be whatever's listed in `[DEFAULT]` (i.e. `2`). Similarly, running with `--profile other-account` would have execute with the API key `987-some-key` and an indentation of `4`, but will revert to the default format of `yaml`.

Reply:
Instead of a ban on having it in the DEFAULT, maybe just adding a check only to the existing setup_module assertion since it should exist either as an env variable or in the jccli-dev-testing section.

e.g. something like:

```python
def setup_module(module):
    api_key = os.getenv('JC_API_KEY')

    if not api_key:
    	# do not use the defaults on this check
    	configparser.ConfigParser(default_section=None)
    	config.read([CONFIG_FILE_PATH])
    	api_key = config['jccli-dev-testing'].get('key')

    assert (api_key is not None), \
        "Either a `JC_API_KEY` environment variable or a `key` in the jccli.ini [jccli-dev-testing] section is required"

```

Reply:
This is a great idea and I implemented it